libsolidity/formal/BMC.cpp

```diff
@@ -151,11 +151,14 @@ void BMC::endVisit(ContractDefinition const& _contract)
 
 bool BMC::visit(FunctionDefinition const& _function)
 {
+	// Free functions need to be visited in the context of a contract.
+	if (!m_currentContract)
+		return false;
+
```

Comment:
This means that free functions, even if they don't actually depend on the context of a contract, will only be analyzed when called, won't it? So a library with only free functions can no longer be SMT-checked?

Also is the function being ``pure`` a sufficient condition for it not to depend on any contract context and, if so, might it make sense to distinguish between pure and non-pure free functions?

Reply:
Currently (before this PR) it is already the case that internal functions are only analyzed if reachable, but that's a fair point we should consider.

Reply:
Actually to be precise: they are encoded and analyzed even if unreachable, but if a property fails inside an internal function, it won't be reported if the function is unreachable.